python/dgl/distributed/kvstore.py

```diff
@@ -0,0 +1,1072 @@
+        partition_book : GraphPartitionBook or RangePartitionBook
+            Store the partition information
+        init_func : func
+            UDF init function
+        """
+        assert len(name) > 0, 'name cannot be empty.'
+        assert len(shape) > 0, 'shape cannot be empty'
+        assert policy_str in ('edge', 'node'), 'policy_str must be \'edge\' or \'node\'.'
+        if self._freeze:
+            raise RuntimeError("KVClient cannot create new \
+                data after invoking get_shared_data() API.")
+        shape = list(shape)
+        if self._client_id == 0:
+            for machine_id in range(self._machine_count):
+                if policy_str == 'edge':
+                    part_dim = len(partition_book.partid2eids(machine_id))
+                elif policy_str == 'node':
+                    part_dim = len(partition_book.partid2nids(machine_id))
+                else:
+                    raise RuntimeError("Cannot support policy: %s" % policy_str)
+                part_shape = shape.copy()
+                part_shape[0] = part_dim
+                request = InitDataRequest(name,
+                                          tuple(part_shape),
+                                          F.reverse_data_type_dict[dtype],
+                                          policy_str,
+                                          init_func)
+                for n in range(self._group_count):
+                    server_id = machine_id * self._group_count + n
+                    rpc.send_request(server_id, request)
+            for _ in range(self._server_count):
+                response = rpc.recv_response()
+                assert response.msg == INIT_MSG
+        self.barrier()
+        # Create local shared-data
+        if policy_str == 'edge':
+            local_dim = len(partition_book.partid2eids(self._machine_id))
+        elif policy_str == 'node':
+            local_dim = len(partition_book.partid2nids(self._machine_id))
+        else:
+            raise RuntimeError("Cannot support policy: %s" % policy_str)
+        local_shape = shape.copy()
+        local_shape[0] = local_dim
+        if self._part_policy.__contains__(name):
+            raise RuntimeError("Policy %s has already exists!" % name)
+        if self._data_store.__contains__(name):
+            raise RuntimeError("Data %s has already exists!" % name)
+        if self._full_data_shape.__contains__(name):
+            raise RuntimeError("Data shape %s has already exists!" % name)
+        self._part_policy[name] = PartitionPolicy(policy_str, self._part_id, partition_book)
+        shared_data = empty_shared_mem(name+'-kvdata-', False, \
+            local_shape, F.reverse_data_type_dict[dtype])
+        dlpack = shared_data.to_dlpack()
+        self._data_store[name] = F.zerocopy_from_dlpack(dlpack)
+        self._data_name_list.add(name)
+        self._full_data_shape[name] = tuple(shape)
+
+    def get_shared_data(self, partition_book):
```

Comment:
maybe call it `map_shared_data`. `get_shared_data` sounds like it will return something.

Reply:
Sure